cpp/log/file_db-inl.h

```diff
@@ -189,6 +192,39 @@ void FileDB<Logged>::RemoveNotifySTHCallback(
 
 
 template <class Logged>
+void FileDB<Logged>::InitializeNode(const std::string& node_id) {
+  CHECK(!node_id.empty());
+  std::unique_lock<std::mutex> lock(lock_);
+  std::string existing_id;
+  if (NodeId(lock, &existing_id) != this->NOT_FOUND) {
+    LOG(FATAL) << "Attempting to initialze DB belonging to node with node_id: "
+               << existing_id;
+  }
+  CHECK(meta_storage_->CreateEntry(kMetaNodeIdKey, node_id).ok());
+}
+
+
+template <class Logged>
+typename Database<Logged>::LookupResult FileDB<Logged>::NodeId(
+    std::string* node_id) {
+  std::unique_lock<std::mutex> lock(lock_);
+  return NodeId(lock, CHECK_NOTNULL(node_id));
+}
+
+
+template <class Logged>
+typename Database<Logged>::LookupResult FileDB<Logged>::NodeId(
+    const std::unique_lock<std::mutex>& lock, std::string* node_id) {
+  CHECK(lock.owns_lock());
```

Comment:
Why do you need that lock? That lock protects the in-memory state of entries (the vector and map), ```FileStorage``` is thread-safe and does not need locking.